duktape/src/main/jni/DuktapeContext.cpp

```diff
@@ -219,3 +228,52 @@ void DuktapeContext::bindInstance(JNIEnv *env, jstring name, jobject object, job
   // Pop the Duktape global object off the stack.
   duk_pop(m_context);
 }
+
+jobject DuktapeContext::call(JNIEnv *env, jstring name, jobject method, jobjectArray args) {
+  duk_push_global_object(m_context);
+  const JString instanceName(env, name);
+  if (!duk_get_prop_string(m_context, -1, instanceName)) {
+    duk_pop(m_context);
+    const jclass illegalArgumentException = env->FindClass("java/lang/IllegalArgumentException");
+    const std::string message = "A global object called " + instanceName.str() + " was not found";
```

Comment:
I think I agree.  My take is that in your example above, if you proxy something to x but then reassign x in JS, the Java proxy still points to the previous instance. (I am still working out how to look up the object/property with something stronger than the name).  

This also applies for another case where you change step 3 above to `x.greet = function() { return "howdy"; };`.  The Java proxy should still be connected to the previous function property that was replaced.